katdal/visdatav4.py

```diff
@@ -555,6 +559,20 @@ def _set_keep(self, time_keep=None, freq_keep=None, corrprod_keep=None,
                 flag_transforms.append(lambda flags: da.bitwise_and(select, flags))
             flag_transforms.append(lambda flags: flags.view(np.bool_))
             self._flags = DaskLazyIndexer(self._corrected.flags, stage1, flag_transforms)
+            unscaled_weights = self._corrected.unscaled_weights
+            if unscaled_weights is None or self.accumulations_per_dump is None:
+                self._excision = None
+            else:
+                # The maximum / expected number of CBF dumps per SDP dump
+                cbf_dumps_per_sdp_dump = int(round(self.dump_period / self.cbf_dump_period))
+                accs_per_sdp_dump = np.float32(self.accumulations_per_dump)
+                accs_per_cbf_dump = accs_per_sdp_dump / np.float32(cbf_dumps_per_sdp_dump)
+                # Each unscaled weight represents the actual number of accumulations per SDP dump.
+                # Correct most of the weight compression artefacts by forcing each weight to be
+                # an integer multiple of CBF n_accs, and then convert it to an excision fraction.
+                excision_transforms = [lambda w: da.round(w / accs_per_cbf_dump) * accs_per_cbf_dump,
+                                       lambda w: (accs_per_sdp_dump - w) / accs_per_sdp_dump]
```

Comment:
I'd rather have a single named function here, so that `DaskLazyIndexer.__str__` will print a meaningful name for the transformation.

Reply:
Cool idea - I should do the same for flags when I add the raw flags machinery. I even had a look at `d._excision.dataset.__dask_graph__().layers.keys()` out of curiosity but I guess that's too low-level.